#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

type_enum = gen.enum([gen.const("centroid", str_t, "centroid", "Use Centroid"),
                      gen.const("initial",  str_t, "initial",  "Use Initial"),
                      gen.const("middle",   str_t, "middle",   "Use Middle")],
                      "An enum to set frontier type")
gen.add("timeout",           double_t, 0, "Explore Timeout(s)", 30.0, 10.0, 100.0)
gen.add("min_frontier_size", double_t, 0, "Min Frontier Size(m)",  0.75,  0.5,   2)
gen.add("visualize",         bool_t,   0, "Is Visualize?",   True)
gen.add("frontier_type",     str_t,    0, "Select Frontier Type",   "centroid", edit_method=type_enum)
gen.add("potential_scale",   double_t, 0, "Distance Weight", 5.0)
gen.add("gain_scale",        double_t, 0, "Gain Weight",     0.5)
gen.add("orientation_scale", double_t, 0, "Orientation Weight",     1.2)
gen.add("information_scale", double_t, 0, "Information Weight",     0.7)
gen.add("information_r",     double_t, 0, "Information Radius",     0.1)

exit(gen.generate("explore_server", "explore_server", "Explore"))
